Network/GenericServerErrorModel.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  WeatherError.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 04/04/2024.
+//
+
+import Foundation
+import Alamofire
+
+// MARK: GenericServerErrorModel
+/// generic server error model
+struct GenericServerErrorModel: LocalizedError, Codable {
+    var cod, message: String?
+    
+    var weatherError: WeatherError?
```

Comment:
```suggestion
    let cod, message: String?
    let weatherError: WeatherError?
```
Please avoid mutable properties when possible.